### ISSUE TITLE
[Suggestion] Add Label support to only extract certain torrents. 

### ISSUE THREAD

**Issue by mmarquezs** [cvarta/deluge-extractor#2]
It would be nice to add label support so it extract all except torrents with certain labels or only extract certain torrents with certain labels.

**Comment by ludwich** [comment]
Thumbs up for this one. Or directory dependent.

**Comment by spreadred** [comment]
Is anyone familiar with how the built-in label plugin works? IE: from inside the extractor plugin, how would we be able to see what the label is?

EDIT: I'm asking so that perhaps I can implement this feature for us.

**Comment by ludwich** [comment]
Never tried the built in since i think that its too limited. I use LabelPlus http://forum.deluge-torrent.org/viewtopic.php?f=9&t=42629

**Comment by seanvree** [comment]
yeah, of all the tools out there, it still blows my mind that no one has created an extract tool based on labels (I.e., if the label is 'Movie' it goes to the 'movie' folder.  Kills me.

**Comment by mmarquezs** [comment]
@seanvree The example you show can already be done. You can use labels, standar one or from the labelplus plugin,  and assign a completed folder for that label, "/completed/movies" . When the movie finishes it will go there and then, if you have the option "Extract torrent in place" set to true in this plugin, it will extract in that folder the files. Unless I'm missing something you can do that already. 

I suggested this idea for when you download some things that don't need to be extracted.
For example I download movies and tv shows that I want to be extracted but if i download a torrent full of NES Roms in zip files for example I don't want it to be extracted.

**Comment by seanvree** [comment]
@mmarquezs  - Yeah, that only works for ONE label since you can't have multiple instances. This only works for ONE label/one target dir. Unless I'm missing something.

**Comment by d8ahazard** [comment]
This sounds interesting, and useful to me. I'll try to implement in my fork for V2. Keep your eyes peeled. 👯‍♂

**Comment by d8ahazard** [comment]
DONE.

This has been committed to my fork of the repository, a release can be found here:

https://github.com/d8ahazard/deluge-extractor/releases/tag/v0.9
